src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -655,31 +655,48 @@ private static FailureDetails GetFailureDetails(Exception e)
             [NotNullWhen(true)] out string? exceptionType,
             [NotNullWhen(true)] out string? exceptionMessage)
         {
-            // Example exception messages:
+            // In certain situations, like when the .NET Isolated worker is configured with
+            // WorkerOptions.EnableUserCodeException = true, the exception message we get from the .NET Isoalted
+            // worker looks like this:
+            // "Exception of type 'ExceptionSerialization.Function+UnknownException' was thrown."
+            const string startMarker = "Exception of type '";
+            const string endMarker = "' was thrown.";
+            if (exception.StartsWith(startMarker) && exception.EndsWith(endMarker))
+            {
+                exceptionType = exception[startMarker.Length..^endMarker.Length];
+                exceptionMessage = string.Empty;
+                return true;
+            }
```

Comment:
not for this PR, but I'd love to revisit this approach - something about string manipulations here makes me unsure about the robustness of our approach

Reply:
Yes, I'm uncomfortable with this too. I sent @mattchenderson a ping on Teams earlier to see if the Functions Host team might consider giving us a proper exception contract between the host and the worker in the future, which would eliminate the need for this kind of fragile string parsing.